include/experimental/__p1673_bits/conjugated.hpp

```diff
@@ -51,74 +51,34 @@ inline namespace __p1673_version_0 {
 namespace linalg {
 
 template<class Accessor>
-class accessor_conjugate;
-
-namespace impl {
-  template<
-    class Accessor,
-    bool is_arith =
-      std::is_arithmetic_v<
-        std::remove_cv_t<typename Accessor::element_type>
-    >
-  >
-  struct accessor_conjugate_aliases {};
-
-  template<class Accessor>
-  struct accessor_conjugate_aliases<Accessor, true>
-  {
-    using reference = typename Accessor::reference;
-    using element_type =
-      std::add_const_t<typename Accessor::element_type>;
-    using data_handle_type = typename Accessor::data_handle_type;
-    using offset_policy = typename Accessor::offset_policy;
-  };
-
-  template<class Accessor>
-  struct accessor_conjugate_aliases<Accessor, false>
-  {
-  private:
-    using accessor_value_type =
-      std::remove_cv_t<typename Accessor::element_type>;
-  public:
-     using reference =
-       conjugated_scalar<typename Accessor::reference,
-			 accessor_value_type>;
-    using element_type =
-      std::add_const_t<typename reference::value_type>;
-    using data_handle_type = typename Accessor::data_handle_type;
-    using offset_policy =
-      accessor_conjugate<typename Accessor::offset_policy>;
-  };
-
-} // namespace impl
-
-template<class Accessor>
-class accessor_conjugate {
+class conjugate_accessor {
 private:
   Accessor accessor_;
-  using aliases = impl::accessor_conjugate_aliases<Accessor>;
 
 public:
-  using reference        = typename aliases::reference;
-  using element_type     = typename aliases::element_type;
-  using data_handle_type = typename aliases::data_handle_type;
-  using offset_policy    = typename aliases::offset_policy;
+  using element_type = decltype(impl::conj_if_needed(std::declval<typename Accessor::element_type>()));
```

Comment:
We need `add_const_t` here, as making the `element_type` const would emphasize that this accessor is always read only.

Reply:
There's another issue here, which is that for the wording change in https://github.com/ORNL/cpp-proposals-pub/pull/401 , the definition of `element_type` is not consistent with the definition of `access`.  I've commented on this below.